src/game.cpp

```diff
@@ -3087,6 +3094,41 @@ void Game::toggleFog(float *statustext_time, bool *flag)
 }
 
 
+void Game::autoPickupKey(float *statustext_time)
+{
+	float toggle_time = g_settings->getFloat("autopickup_key_toggle_time");
+	f32 pickup_range;
+	if (!client->getEnv().autoPickupEnabled()) {
+		bool noclip = client->checkPrivilege("fly")
+			&& g_settings->getBool("free_move")
+			&& client->checkPrivilege("noclip")
+			&& g_settings->getBool("noclip");
+		bool fast = client->checkPrivilege("fly")
+			&& g_settings->getBool("free_move")
+			&& g_settings->getBool("always_fly_fast");
+		client->getEnv().toggleAutoPickup(&pickup_range);
+		statustext = std::wstring(L"Auto pick-up enabled; pick-up range is ")
+			+ utf8_to_wide(ftos(int(2 * pickup_range) / 2.0)) + L" block(s)";
```

Comment:
Blocks are 16x16x16 nodes. We do not want to confuse the players - using meters `m` would be a better choice IMO.